packages/stylelint-plugin/src/rules/no-invalid-use-of-var-tokens-in-functions.ts

```diff
@@ -69,51 +74,46 @@ const parsePercentage = (value: string): number | "NaN" => {
 }
 
 /*
-  Return a replacement rgb-params variable
+  Return a replacement rgb variable
 */
 const getAndReportOnReplacementRgbParamsVariable = (
   variable: Variable,
   decl: Declaration,
   options: Options
 ) => {
   const variablePrefix = variablePrefixForLanguage(options.language)
+
+  // We only care about Kaizen tokens.
   if (!variable.kaizenToken) return undefined
 
-  if (!variable.kaizenToken.cssVariable) {
-    const fixedVariable =
-      kaizenTokensByName[
-        variable.kaizenToken.name.replace("kz", "kz-var") + "-rgb-params"
-      ]
-    if (!fixedVariable) {
-      options.reporter({
-        message: noMatchingRgbParamsVariableMessage(variable.kaizenToken.name),
-        node: decl,
-        autofixAvailable: false,
-      })
-      return undefined
-    }
-    if (options.fix && !isVariable(decl)) {
-      return `${variablePrefix}${fixedVariable.name}`
-    } else {
-      options.reporter({
-        message: invalidRgbaUsage(fixedVariable.name),
-        node: decl,
-        autofixAvailable: !isVariable(decl),
-      })
-    }
+  // We'll decide on what this variable should be using a series of rules, then use it as the subject for making replacements
+  let kaizenTokenSubject: CurrentKaizenToken
+
+  const replacement = getReplacementForDeprecatedOrRemovedToken(variable.name)
+  if (variable.kaizenToken.removed && replacement) {
+    kaizenTokenSubject = replacement
+  } else if (variable.kaizenToken.deprecated && replacement) {
+    kaizenTokenSubject = replacement
+  } else if (
+    !variable.kaizenToken.deprecated &&
+    !variable.kaizenToken.removed
+  ) {
+    kaizenTokenSubject = variable.kaizenToken
+  } else {
+    // Just bail if we reach here, because we couldn't decide on a kaizen token to use.
+    return undefined
```

Comment:
This relates to the difficult separation of concerns within this rule and the `no-deprecated-or-removed-tokens` rule.
I've tried to make them only report on what they're responsible for, and this final condition of there not being a replacement for deprecated/removed tokens is not this rule's responsibility to report on anything.

Reply:
I'll change this a bit to clear up the confusion

Reply:
Yeah the blurring of those rules is unfortunate.

I can't remember - did you previously try an approach to have the "no-deprecated-or-removed-tokens" rule do a naive find/replace in the first pass, and then have the "no-invalid-use-in-functions" rule detect where it ended up with something bad?

I feel like the reason we didn't do that before is because using the non-css-variable tokens in those functions remained an option. But soon we won't have those, so I wonder if keeping the rules more separate is possible.